resource_control: add new privilege `RESOURCE_GROUP_USER` (#17965)
pingcap/docs: sql-statements/sql-statement-set-resource-group.md

|   "DEFAULT"
```

## Privilege

> This statement requires following configuration and privilege:

Reviewer: ```suggestion
Executing this statement requires the following configuration and privilege:
```

> 1. System variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control--new-in-v660) is set to `ON`.
> 2. When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, privilege `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` is needed; when set to `OFF`, none of these privileges are needed.

Reviewer: ```suggestion
1. The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
2. When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.
```

> 1. The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
> 2. When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.

Reviewer 2: ```suggestion
- The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
- When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.
```

Reviewer: The numbered list is used on purpose, to emphasize that both the two must be satisfied.
